WooCommerce/Classes/ViewRelated/JetpackSetup/AdminRole/AdminRoleRequiredView.swift

```diff
@@ -0,0 +1,160 @@
+import SwiftUI
+
+/// Hosting controller that wraps the `AdminRoleRequiredView`.
+final class AdminRoleRequiredHostingController: UIHostingController<AdminRoleRequiredView> {
+    private lazy var noticePresenter: DefaultNoticePresenter = {
+        let noticePresenter = DefaultNoticePresenter()
+        noticePresenter.presentingViewController = self
+        return noticePresenter
+    }()
+
+    init(siteID: Int64, onClose: @escaping () -> Void, onSuccess: @escaping () -> Void) {
+        super.init(rootView: AdminRoleRequiredView(viewModel: .init(siteID: siteID),
+                                                   onClose: onClose,
+                                                   onSuccess: onSuccess))
+        rootView.onShowingNotice = { [weak self] message in
+            self?.displayNotice(message: message)
+        }
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTransparentNavigationBar()
+    }
+
+    private func displayNotice(message: String) {
+        let notice = Notice(title: message)
+        noticePresenter.enqueue(notice: notice)
+    }
+}
+
+/// Error view displayed when a user without admin role tries to install Jetpack.
+struct AdminRoleRequiredView: View {
+    /// Triggered when the user taps Cancel.
+    let onClose: () -> Void
+    /// Triggered when the user taps Retry and their role is updated with the admin role.
+    let onSuccess: () -> Void
+    /// Triggered when an error needs to be displayed.
+    var onShowingNotice: (String) -> Void = { _ in }
+
+    private let viewModel: AdminRoleRequiredViewModel
+
+    /// Provides the URL destination when the link button is tapped
+    private let linkDestinationURL = WooConstants.URLs.rolesAndPermissionsInfo.asURL()
```

Comment:
Nit: Can we consider moving this to `AdminRoleRequiredViewModel`?

Reply:
I'm keeping this on the view since it doesn't involve any logic update, and we will not add tests for it.